modules/javafx.controls/src/main/java/javafx/scene/control/skin/TreeTableRowSkin.java

```diff
@@ -229,30 +213,19 @@ public final DoubleProperty indentProperty() {
         super.updateChildren();
 
         updateDisclosureNodeAndGraphic();
-
-        if (childrenDirty) {
-            childrenDirty = false;
-            if (cells.isEmpty()) {
-                getChildren().clear();
-            } else {
-                // TODO we can optimise this by only showing cells that are
-                // visible based on the table width and the amount of horizontal
-                // scrolling.
-                getChildren().addAll(cells);
-            }
-        }
+        disclosureNodeDirty = false;
     }
 
     /** {@inheritDoc} */
     @Override protected void layoutChildren(double x, double y, double w, double h) {
-        if (disclosureNodeDirty) {
-            updateDisclosureNodeAndGraphic();
-            disclosureNodeDirty = false;
+        Node disclosureNode = getDisclosureNode();
+        if (disclosureNode != null && disclosureNode.getParent() == null) {
+            disclosureNodeDirty = true;
         }
 
-        Node disclosureNode = getDisclosureNode();
-        if (disclosureNode != null && disclosureNode.getScene() == null) {
+        if (disclosureNodeDirty) {
             updateDisclosureNodeAndGraphic();
+            disclosureNodeDirty = false;
```

Comment:
shouldn't `disclosureNodeDirty = false;` be moved to `updateDisclosureNodeAndGraphic();` ?
here and in L257, L216

Reply:
Could be an idea. I will check. I just tried to not make to many changes, but this might be an easy one.

Reply:
Checked the code, that is a safe improvement, so will do.